pkg/runtime/wfengine/activity.go

```diff
@@ -175,7 +176,16 @@ func (a *activityActor) executeActivity(ctx context.Context, actorID string, nam
 		}
 		return newRecoverableError(fmt.Errorf("failed to schedule an activity execution: %w", err))
 	}
-
+	// Activity execution started
+	start := time.Now()
+	executionStatus := ""
+	elapsed := float64(0)
+	// Record metrics on exit
+	defer func() {
+		if executionStatus != "" {
+			diag.DefaultWorkflowMonitoring.ExecutionEvent(ctx, diag.ComponentName, diag.Activity, executionStatus, elapsed)
+		}
+	}()
```

Comment:
adding a test verifying the different execution status values on return would be good. It can be UT or IT.

Reply:
The general UTs are there as similar to other metrics. IT for workflow is not there yet, more tests can be added there as required. Can that be a separate item to add IT for workflow?

Reply:
Added ITs as well